adobe/spectrum-css
# docs: deprecation pages and styles fixed

## Description

A few patches to our existing docs site to render our deprecated pages and load the deprecated styles appropriately.

### Detailed updates

- Fixed a bug in the splitbutton documentation missing the iconOnly class for buttons
- Remove legacy vars & expressvars packages
- Add missing checkbox peer dependency in card (required to bring it's styles into the docs page)
- Example background color was using gray-50 which is too dark for disabled slider elements; updated to gray-100
- Include deprecated pages and styles in the build process
- Remove polyfills.js as outdated
- Remove halo-focus, truncation, and wrapping pages that aren't surfaced to the site publicly
- Remove unused individualComponent template

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

Open the [docs site](https://pr-2611--spectrum-css.netlify.app/) @mdt2 
  - [x] Landing page loads all images
  - [x] Navigation now includes both Components and Deprecated groupings
  - [x] The component preview link should be there and link to storybook


Pick 3 random Component pages. Do they: @mdt2
  - [x] Respond to the context switcher (light/dark? medium/large?) Do those settings persist when you change pages?
  - [x] Interactions still work on the components as they did previously? (Do popovers still open? Dialogs close? etc.)
  - [x] Do styles match the current site and are rendering as expected? (Expect to see a lighter example background however.)

Check the deprecated pages. Do they: @mdt2 (checked on docs site and storybook)
  - [x] Show a deprecated notice at the top?
  - [x] Respond to the context switcher?
  - [x] Interactions still work on the components as they did previously?
  - [x] Do styles render as expected?

## To-do list
